pulpcore/tests/unit/serializers/test_repository.py

```diff
@@ -167,3 +175,83 @@ def test_validate_repo_remote_sync(monkeypatch):
     error_msg = r"Type for Remote .* does not match Repository .*."
     with pytest.raises(serializers.ValidationError, match=error_msg):
         serializer.validate(data)
+
+
+@pytest.mark.parametrize("tested_serializer", [PublicationSerializer(), DistributionSerializer()])
+def test_validate_repo_remote_via_repository(monkeypatch, tested_serializer):
+    """
+    Test that the publication/distribution fails if
+    "repository" has a remote of an incompatible type.
+    """
+    monkeypatch.setattr(ValidateFieldsMixin, "CHECK_SAME_DOMAIN", False)
```

Comment:
I'd rather patch `check_cross_domains` to `lambda ...: None`. I had to look this one up to understand.
Would that work?
```suggestion
    monkeypatch.setattr(tested_serializer, "check_cross_domain", lambda self, data: None)
```